RUFAS/input_manager.py

```diff
@@ -300,6 +343,214 @@ def _filter_input_data_by_metadata(self, input_data: Dict[str, Any],
+            A dictionary containing the properties of the variable, including details that determine its modifiability.
+
+        Returns
+        -------
+        bool
+            True if the variable is allowed to be modified during runtime, False otherwise.
+
+        Notes
+        -----
+        - The function relies on the 'Modifiability' enum and the '_get_variable_modifiability' method. Changes to these
+          components may impact the function's behavior.
+        - This function does not consider other aspects of the variable that might affect its ability to be modified at
+          runtime, focusing solely on its declared modifiability status.
+        """
+        variable_modifiability = self._get_variable_modifiability(variable_name=variable_name,
+                                                                  variable_properties=variable_properties)
+        return (variable_modifiability == Modifiability.NOT_REQUIRED_AND_UNLOCKED) or \
+               (variable_modifiability == variable_modifiability.REQUIRED_AND_UNLOCKED)
+
+    def _handle_missing_data(self, variable_properties: Dict[str, Any], var_name: str) \
+            -> None:
+        """
+        Handles missing data for a variable, logging errors or warnings based on the context of initialization or
+        runtime updates.
+
+        Parameters
+        ----------
+        variable_properties : Dict[str, Any]
+            Properties of the variable, potentially including its modifiability status.
+        var_name : str
+            The name of the variable with missing data.
+
+        Raises
+        ------
+        KeyError
+            Raised if the missing data is deemed necessary, either during initialization or for a runtime update.
+
+        Notes
+        -----
+        This function determines if it's being called during the initialization phase and checks if the missing variable
+        data is required at this stage using '_is_input_required_upon_initialization'. If required, it logs an error and
+        raises a KeyError. If not, it logs a warning.
+
+        For missing data during runtime updates, it logs an error and raises a KeyError, assuming the data is necessary
+        for the update.
+
+        - Relies on the caller function's name to determine if it's called during initialization, making the function
+          sensitive to naming conventions of the initialization method.
+        """
+        caller_function = self._get_caller_function()
+        is_initialization = caller_function == self._populate_pool.__name__
+
+        info_map = {
+            "class": self.__class__.__name__,
+            "function": caller_function
+        }
+        if not is_initialization:
+            om.add_error(
+                "Validation: key not found in data -- value required to update variable during runtime",
+                f"Key {var_name} not found in data. A value is required for to update variable during runtime.",
```

Comment:
```suggestion
                "missing required data",
                f"Key {var_name} not found in data. A value is required for to update variable during runtime.",
```
don't repeat yourself.